sale_order_lot_selection/model/stock.py

```diff
@@ -11,7 +11,7 @@ class StockMove(models.Model):
     _inherit = 'stock.move'
 
     @api.model
-    def _prepare_procurement_from_move(self, move):
-        vals = super(StockMove, self)._prepare_procurement_from_move(move)
-        vals['lot_id'] = move.restrict_lot_id.id
+    def _prepare_procurement_from_move(self):
+        vals = super(StockMove, self)._prepare_procurement_from_move()
+        vals['lot_id'] = self.restrict_lot_id.id
```

Comment:
It may work this way, but it is confusing, usually we use `@api.model` when we don't care the value of `self`. But here, we need `self.restrict_lot_id.id`, so we actually care about the value.
That is why api.model is not good in this case.